src/app/pages/quickMaps/pages/baselineDetails/mapView/mapView.component.ts

```diff
@@ -273,47 +384,107 @@ export class MapViewComponent implements AfterViewInit {
     this.thresholdLegend.addTo(this.thresholdMap);
   }
 
-
   private populateFeatureCollection(data: Array<SubRegionDataItem>): void {
     this.areaFeatureCollection = {
       type: 'FeatureCollection',
-      features: data
-        .map((item: SubRegionDataItem) => item.geoFeature)
-        .filter((item) => null != item),
+      features: data.map((item: SubRegionDataItem) => item.geoFeature).filter((item) => null != item),
     };
   }
 
-  private getColourAbsolute(absoluteValue: number): string {
-    return absoluteValue > 1500
-      ? '#2ca25f'
-      : absoluteValue > 1000
-        ? '#addd8e'
-        : absoluteValue > 500
-          ? '#ffeda0'
-          : absoluteValue > 250
-            ? '#feb24c'
-            : absoluteValue > 100
-              ? '#f03b20'
-              : absoluteValue > 50
-                ? '#bd0026'
-                : absoluteValue > 10
-                  ? '#7a0177'
-                  : '#354969';
-  }
-  private getColourThreshold(thresholdValue: number): string {
-    return thresholdValue > 99
-      ? '#187026'
-      : thresholdValue > 80
-        ? '#22f243'
-        : thresholdValue > 60
-          ? '#f0f01d'
-          : thresholdValue > 40
-            ? '#f08f11'
-            : thresholdValue > 20
-              ? '#e0071c'
-              : thresholdValue > 0
-                ? '#5e1d5c'
-                : '#0f0e0f';
+  private getColourRange(absoluteValue: number, thresholdValue: number, colourGradient: ColourGradientType): string {
```

Comment:
Might be easier to have two parameters here as range (an enum that is "absolute" or "threshold", which is a bit like how you've done the "colourGradient") and "value" for the number value, instead of two variables that only one is populated.